File: Sis104Avance/app/src/main/java/com/example/sis104avance/ejercicios/EcuacionActivity.java
package com.example.sis104avance.ejercicios;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.example.sis104avance.MainActivity;
import com.example.sis104avance.R;

public class EcuacionActivity extends AppCompatActivity {

    EditText editTextA, editTextB, editTextC;
    Button button_SolucionEcuacion,button_salir;
    TextView textViewResultado;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_ecuacion);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        editTextA = findViewById(R.id.editTextA);
        editTextB = findViewById(R.id.editTextB);
        editTextC = findViewById(R.id.editTextC);
        button_SolucionEcuacion = findViewById(R.id.button_ResolverEcuacion2doGrado);
        button_salir = findViewById(R.id.button_Salir_Ecuacion2doGrado);
        textViewResultado = findViewById(R.id.textViewResultado_Ecuacion2doGrado);

        button_SolucionEcuacion.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v){
                double a = Double.parseDouble(editTextA.getText().toString());
                double b = Double.parseDouble(editTextB.getText().toString());
                double c = Double.parseDouble(editTextC.getText().toString());
                Ecuacion ecuacion = new Ecuacion(a,b,c);
                textViewResultado.setText(ecuacion.Raices());
            }
        });

        button_salir.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(EcuacionActivity.this, EjerciciosActivity.class);
                startActivity(intent);
            }
        });

    }
}

File: trabajoSIS104/app/src/main/java/com/example/trabajosis104/ejercicios/EcuacionActivity.kt
package com.example.trabajosis104.ejercicios

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.trabajosis104.R

class EcuacionActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_ecuacion)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        // Casting explícito al tipo adecuado
        val editTextA = findViewById<EditText>(R.id.editTextA)
        val editTextB = findViewById<EditText>(R.id.editTextB)
        val editTextC = findViewById<EditText>(R.id.editTextTextC)
        val textViewResultadoEcuacion = findViewById<TextView>(R.id.textViewResultadoEcuacion)
        val buttonResolverEcuacion = findViewById<Button>(R.id.buttonResolverEcuacion)
        val buttonSalirEcuacion = findViewById<Button>(R.id.buttonSalirEcuacion)

        buttonResolverEcuacion.setOnClickListener {
            // Lógica para resolver la ecuación
            val valorA = editTextA.text.toString().toDoubleOrNull()
            val valorB = editTextB.text.toString().toDoubleOrNull()
            val valorC = editTextC.text.toString().toDoubleOrNull()

            if (valorA == null || valorB == null || valorC == null) {
                val errores = mutableListOf<String>()

                if (valorA == null) {
                    errores.add("A")
                }
                if (valorB == null) {
                    errores.add("B")
                }
                if (valorC == null) {
                    errores.add("C")
                }

                textViewResultadoEcuacion.text = "Por favor, ingrese valores válidos en: ${errores.joinToString(", ")}"
                //Toast.makeText(this, "Valores inválidos: ${errores.joinToString(", ")}", Toast.LENGTH_SHORT).show()
            } else {
                // Si los valores son válidos, resolver la ecuación
                val ecuacion = Ecuacion(valorA, valorB, valorC)
                val resultado = ecuacion.calcularRaices()

                textViewResultadoEcuacion.text = resultado
                Toast.makeText(this, resultado, Toast.LENGTH_SHORT).show()
            }
        }

        buttonSalirEcuacion.setOnClickListener {
            finish() // Finaliza la actividad y cierra la pantalla actual
        }
    }
}
